recipes-graphics/wayland/weston-init/profile

```diff
@@ -1,9 +1,5 @@
 #!/bin/sh
 
 # Set XDG_RUNTIME_DIR manually.
-# The variable is not set for ssh login, causing app failures. It's not clear
-# if this is an error or by design, but setting it manually does help and
-# doesn't seem to hurt...
-if test -z "$XDG_RUNTIME_DIR"; then
```

Comment:
The assignment here is a workaround in the case that `XDG_RUNTIME_DIR` is not already set. It's best to limit our assignment to just this case..

Reply:
Yes, our patches do not do the exact same thing.
In your case, you were trying to populate the `XDG_RUNTIME_DIR` if (and only if) not populated, I haven't seen that issue actually.
On my end, the main issue is that `XDG_RUNTIME_DIR` is set... but to the wrong value. It seems to be set by someone but to `id -u` value which is wrong as it should always be `id weston -u` as the service only starts with `weston` user. So my patch makes it so the variable is properly set no matter which user you are using.

Reply:
Oh, that reminds me, I believe your patch violates the spec which says that `XDG_RUNTIME_DIR` is user-specific.

https://wiki.archlinux.org/title/XDG_Base_Directory

Can you switch user to weston?

Reply:
Thanks, that explains what I see, now I know it's `pam_systemd` that sets it to `/run/user/$UID`.
However I think it's just bad not to use the weston user at all time as it simply weston unusable for any user other than weston.
BTW, no `su weston` doesn't work, however if I connect through `ssh weston@` it works ok.

Reply:
Right, I'm not saying there's not necessarily a problem, just that violating the XDG spec seems an indication of something missing in the analysis. Can you ask the weston experts?